src/histolab/tiler.py

```diff
@@ -296,17 +300,59 @@ def tile_size(self, tile_size_: Tuple[int, int]):
 
     # ------- implementation helpers -------
 
+    def _are_coordinates_within_extraction_mask(
+        self,
+        tile_thumb_coords: CoordinatePair,
+        binary_mask_region: np.ndarray,
+    ) -> bool:
+        """Chack whether the ``tile_thumb_coords`` are inside of ``binary_mask_region``.
+
+        Return True if the tile defined by ``tile_thumb_coords`` is inside of the True
+        areas of ``binary_mask_region`` for at least 80% of its area.
+
+        Parameters
+        ----------
+        tile_thumb_coords : CoordinatePair
+            Coordinates of the tile at thumbnail dimension.
+        binary_mask_region : np.ndarray
+            Binary mask with True inside of the tissue region considered.
+
+        Returns
+        -------
+        bool
+            Whether the tile is inside of the True areas of ``binary_mask_region`` for
+            at least 80% of its area.
+        """
+
+        tile_thumb_mask = rectangle_to_mask(
+            dims=binary_mask_region.shape, vertices=tile_thumb_coords
+        )
+
+        tile_in_binary_mask = binary_mask_region & tile_thumb_mask
+
+        tile_area = np.count_nonzero(tile_thumb_mask)
+        tile_in_binary_mask_area = np.count_nonzero(tile_in_binary_mask)
+
+        return (
+            tile_in_binary_mask_area / tile_area > COORDS_WITHIN_EXTRACTION_MASK_THRESH
+        )
+
     def _grid_coordinates_from_bbox_coordinates(
-        self, bbox_coordinates: CoordinatePair, slide: Slide
+        self,
+        bbox_coordinates_lvl: CoordinatePair,
+        slide: Slide,
+        binary_mask_region: np.ndarray,
     ) -> CoordinatePair:
         """Generate Coordinates at level 0 of grid tiles within a tissue box.
 
         Parameters
         ----------
-        bbox_coordinates: CoordinatePair
+        bbox_coordinates_lvl : CoordinatePair
             Coordinates of the tissue box from which to calculate the coordinates.
         slide : Slide
             Slide from which to calculate the coordinates.
+        binary_mask_region : np.ndarray
+            Binary mask corresponding to the region considered
```

Comment:
"the region considered" is not that clear

Reply:
ok changed and added a Notes section for clarity